web/src/ui/timeline/TimelineView.tsx

```diff
@@ -13,38 +13,89 @@
 //
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
-import { use, useCallback, useEffect, useLayoutEffect, useRef, useState } from "react"
+import { Virtualizer, useVirtualizer } from "@tanstack/react-virtual"
+import { use, useEffect, useLayoutEffect, useRef, useState } from "react"
 import { ScaleLoader } from "react-spinners"
 import { usePreference, useRoomTimeline } from "@/api/statestore"
 import { EventRowID, MemDBEvent } from "@/api/types"
 import useFocus from "@/util/focus.ts"
 import ClientContext from "../ClientContext.ts"
 import { useRoomContext } from "../roomview/roomcontext.ts"
 import TimelineEvent from "./TimelineEvent.tsx"
+import { getBodyType, isSmallEvent } from "./content/index.ts"
 import "./TimelineView.css"
 
+const measureElement = (element: Element, entry: ResizeObserverEntry | undefined, instance: Virtualizer<HTMLDivElement, Element>) => {
+	const horizontal = instance.options.horizontal
+	const style = window.getComputedStyle(element)
+	if (entry == null ? void 0 : entry.borderBoxSize) {
+		const box = entry?.borderBoxSize[0]
+		if (box) {
+			const size = Math.round(
+				box[horizontal ? "inlineSize" : "blockSize"],
+			)
+			return size + parseFloat(style[horizontal ? "marginInlineStart" : "marginBlockStart"]) + parseFloat(style[horizontal ? "marginInlineEnd" : "marginBlockEnd"])
```

Comment:
There are multiple lines that are too long (I guess I should make that eslint rule an error 🤔)